packages/sdk-btc/source/transaction.service.ts

```diff
@@ -130,19 +145,25 @@ export class TransactionService extends Services.AbstractTransactionService {
 				};
 			} else if (levels.purpose === 49) {
 				let network = getNetworkConfig(this.configRepository);
+
 				const payment = bitcoin.payments.p2sh({
 					redeem: bitcoin.payments.p2wpkh({
 						pubkey: Buffer.from(signingKey.publicKey, "hex"),
 						network,
 					}),
 					network,
 				});
+
```

Comment:
@faustbrian this is not possible really, that why I was using `!`. The payment is created in the previous statement and we're setting the redeem field explicitly. So even though it's optional in the structure, it should always be present.

Reply:
I understand that but I prefer to have a safeguard here if the package itself declares this as optional, even if it's just a faulty type declaration.